src/domain/application/hooks/useSaveApplication.ts

```diff
@@ -20,15 +21,18 @@ export default function useSaveApplication<
     null,
   );
 
-  const applicationCreateMutation = useMutation(createApplication<ApplicationData, CreateData>, {
-    onMutate() {
-      setShowSaveNotification(null);
-    },
-    onSuccess: onCreateSuccess,
-    onSettled() {
-      setShowSaveNotification('create');
+  const applicationCreateMutation = useDebouncedMutation(
+    createApplication<ApplicationData, CreateData>,
+    {
+      onMutate() {
+        setShowSaveNotification(null);
+      },
+      onSuccess: onCreateSuccess,
+      onSettled() {
+        setShowSaveNotification('create');
+      },
     },
-  });
+  );
 
   const applicationUpdateMutation = useMutation(updateApplication<ApplicationData, UpdateData>, {
```

Comment:
Should this (update application) also use `useDebouncedMutation`? There was some duplicate key errors in logs from application update.

Reply:
Good point. Changed it so that update also uses `useDebouncedMutation`.